app/src/main/java/com/example/hangouts/homeScreen/ActiveHangoutsAdapter.java

```diff
@@ -59,14 +61,14 @@ public ViewHolder(@NonNull ItemActiveHangoutBinding binding) {
         }
 
         public void bind(Hangout hangout) {
-            try{
-                hangout = hangout.fetchIfNeeded();
-            }catch (ParseException e){
-                e.printStackTrace();
-            }
-            binding.tvItemAlias.setText(hangout.getAlias());
-            binding.tvItemLocation.setText(hangout.getLocationString());
-            binding.tvItemMembers.setText(String.valueOf(hangout.getMembers().length()));
+            hangout.fetchIfNeededInBackground(new GetCallback<Hangout>() {
```

Comment:
I was wondering why when you had many items in the recycler view the scroll felt glitchy. I was fetching the data from the hangouts synchronically instead of in the background.

Reply:
That would certainly do it!
is `fetchIfNeededInBackground` a method provided by parse?  guessing it checks a cache and only fetches if it is a miss?